lib/vector-core/buffers/src/disk/leveldb_buffer/reader.rs

```diff
@@ -135,6 +139,7 @@ where
 {
     fn drop(&mut self) {
         self.delete_acked();
+        self.flush(self.current_size.load(Ordering::Acquire));
```

Comment:
I was thinking about whether you actually need this Acquire/Release pair but the more I think about it the more I've come around to the notion that we could avoid sequencing entirely by having `delete_acked` return the `unread_size` which we then pass directly into `flush`. Might even be able to drop `current_size` down to a `usize` at that point.

Reply:
> delete_acked return the unread_size 

That seems like a good solution. I'll add it.

> Might even be able to drop current_size down to a usize at that point.

Writers are using it also so nope.

Reply:
Done. This also moves flush to after we wake writers which should be a throughput win in some scenarios. I've also done sanity run on how this move interacts with other things and it seems fine.